components/UI/InviteUsersToMatrixRoom/index.js

```diff
@@ -0,0 +1,126 @@
+    const auth = useAuth();
+    const matrixClient = auth.getAuthenticationProvider('matrix').getMatrixClient();
+    const [searchResults, setSearchResults] = useState([]);
+    const [userFeedback, setUserFeedback] = useState('');
+    const [errorFeedback, setErrorFeedback] = useState([]);
+    const { t } = useTranslation('invitationModal');
+
+    const handleChange = (searchString) => {
+        debouncedFetchUsersForContributorSearch(searchString);
+    };
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    const debouncedFetchUsersForContributorSearch = useCallback(debounce((val) => fetchUsersForContributorSearch(val), 300), []);
+
+    const fetchUsersForContributorSearch = useCallback(async (a) => {
+        try {
+            const users = await matrixClient.searchUserDirectory({ term: a });
+            // always filter ourselves; we most likely do not want to invite ourselves to something, i guess?!
+            const usersWithoutMyself = _.filter(users.results, (user) => user.user_id !== matrixClient.getUserId());
+            // we only update the state if the returned array has entries, to be able to check if users a matrix users or not further down in the code (otherwise the array gets set to [] as soon as you selected an option from the datalist)
+            // const filterResults = users.results.filter(item => _.isEqual(item, option));
+            setSearchResults(usersWithoutMyself);
+        } catch (err) {
+            logger.error(t('Error while trying to fetch users: ') + err);
+        }
+    }, [matrixClient, t]);
+
+    function clearInputs() {
+        setUserFeedback('');
+        setSearchResults([]);
+    }
+
+    const handleInvite = async (selectedUsers) => {
+        setErrorFeedback([]);
+        const errors = [];
+
+        for (const user of selectedUsers) {
+            await matrixClient.invite(roomId, user.user_id)
+                .catch(async error => {
+                    // avoid adding duplicates
+                    if (errors.includes(error.data.error)) return;
+                    errors.push(error.data.error);
+                });
+        }
+
+        if (errors.length !== 0) {
+            // if something went wrong we display the errors and clear all inputs
+            setErrorFeedback(errors);
+        }
+
+        const successAmount = selectedUsers.length - errors.length;
+
+        // if everything is okay, we let the user know and exit the view.
+        successAmount > 0 && setUserFeedback('✓ ' + successAmount + ' ' + t('{{user}} invited and {{needs}} to accept your invitation', { user: successAmount > 1 ? 'users were' : 'user was', needs: successAmount > 1 ? 'needs' : 'need' }));
+        await new Promise(() => setTimeout(() => {
+            clearInputs();
+            if (onSuccess && successAmount === selectedUsers.length) onSuccess();
+        }, 3000));
+    };
+
+    return <ActionWrapper>
+        <h3>{ t('Invite users') }</h3>
```

Comment:
Missing translations for this one, and even more. The button also says "Invite" in the German version.

Reply:
I've added a <Trans /> component for the more complicated translation now with commit b62f64ab9e8090de341f8b37971e126b1c7137ff. Improves readability and also makes potential future translations easier. Had to add an additiona `en/invitationModal.json` file for this, tho.